app/scripts/components/misc/spinner-modal.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { PropTypes } from 'react';
+import T from './t';
+
+require('stylesheets/misc/spinner-modal');
+
+const SpinnerModal = React.createClass({
+  propTypes: {
+    retry: PropTypes.func,
+    state: PropTypes.object.isRequired,
+  },
+  render() {
+    return this.props.state.match({
```

Comment:
:smile: Any `Enum` from `results`, including the built-in `Ok`/`Err` and `Some`/`None` pairs.

There's a catch-all too, `_`, so this could have been

```jsx
this.props.state.match({
  Finished: () => <div style={{display: 'none'}}></div>,
  _:  () => <div>Active and Failed would both render this</div>
});
```